Library/Homebrew/formula_creator.rb

```diff
@@ -31,24 +31,38 @@ def verify
       raise TapUnavailableError, @tap.name unless @tap.installed?
     end
 
-    def parse_url
-      path = Pathname.new(@url)
-      if @name.nil?
-        case @url
-        when %r{github\.com/(\S+)/(\S+)\.git}
-          @user = Regexp.last_match(1)
-          @name = Regexp.last_match(2)
-          @head = true
-          @github = true
-        when %r{github\.com/(\S+)/(\S+)/(archive|releases)/}
-          @user = Regexp.last_match(1)
-          @name = Regexp.last_match(2)
-          @github = true
-        else
-          @name = path.basename.to_s[/(.*?)[-_.]?#{Regexp.escape(path.version.to_s)}/, 1]
-        end
+    def self.name_from_url(url)
+      odebug "name_from_url: #{url}"
+      stem = Pathname.new(url).stem
+      odebug "stem: #{stem}"
+      # special cases first
+      name = if stem.start_with? "index.cgi"
+        # parsing gitweb URLs http://www.codesrc.com/gitweb/index.cgi?p=libzipper.git;a=summary
```

Comment:
I wonder if it's specifically worth checking for the domain name here or more bits e.g. `gitweb`?

Reply:
That's a default Git interface https://wiki.archlinux.org/title/gitweb so checking domain name would exclude other installations.

Checking for `gitweb` specifically.. well, the `stem` variable is still needed for other rules, so the would be a bit redundant. I don't think there are many repo browser installations with exposed `index.cgi` which are not `gitweb`. In any case, if somebody reports new URL, it is easy to patch given we've got something to test against.